spec/system/views/posts/index_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'rails_helper'
+
+RSpec.describe 'Post#Index', type: :system do
+  describe 'The Post#Index page' do
+    it 'displays the users profile picture' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_css("img[src*='#{user.photo}']")
+      end
+    end
+
+    it 'displays the username' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_content(user.name)
+      end
+    end
+
+    it 'displays the number of posts' do
+      User.all.each do |user|
+        visit user_posts_path(user.id)
+        expect(page).to have_content(user.posts.count)
+      end
+    end
+
+    it 'displays the posts title' do
```

Comment:
- [ ] Please make sure to include the below test for the posts index page as requested in the project requirements.

![image](https://github.com/Rhaegar121/Blog-app/assets/83803414/491a8276-3d00-46f3-be31-1b7208eee2a1)